woocommerce/woocommerce-ios
# Lock Screen Widgets: Add 3 stats widgets

Depends on: #7805.

# Description
This PR adds 3 new lock screen widgets that are using existing data model.
All widgets are presenting single metric - total revenue for today.

# How

- Updates `CurrencyFormatter` to add `formatHumanReadableAmount` func overloads.
- Adds `revenueCompact` to `StoreInfoData`.
- Adds 3 independent widgets views and connects them to `StoreInfoWidgetEntryView`.

# Testing Steps

1. Use iOS 16 device/simulator.
2. Build and run the app in debug mode (to ensure the feature flag is enabled).
3. Long press on lock screen.
4. Tap "Customize".
5. Tap on a top line to choose inline widget.
6. In "Add Widgets" panel scroll to "Woo" app section and tap on a single option.
7. Tap on a bottom line of widgets to choose circular/rectangular widgets.
8. In "Add Widgets" panel scroll to "Woo" app and tap it.
9. Tap 2 "Today" widgets to add it on the lock screen.
10. Close widgets panel, tap "Done" in top right corner.
11. Check that all widgets on a lock screen are displaying same and correct data.
12. Tap on any newly added widgets and confirm it launches the parent app.

Bonus: try app on iOS 15 device and verify it works (without any new widgets).

# Screenshots

Lock Screen | Widgets Configuration
--|--
![IMG_2448](https://user-images.githubusercontent.com/3132438/193354921-9b4d1cd6-845d-4eb4-a909-57484c27b64a.jpeg)|![IMG_2449](https://user-images.githubusercontent.com/3132438/193354834-9201b9f1-9d4d-497a-b02f-ac8826a61b20.jpeg)

---
